src/Components/ADT3DSceneBuilder/Internal/VisualRuleForm/Internal/ConditionsCallout/ConditionsCallout.tsx

```diff
@@ -0,0 +1,241 @@
+import React, { useCallback, useEffect, useReducer, useRef } from 'react';
+import {
+    ConditionCalloutActionType,
+    ConditionCalloutReducerType,
+    ConditionValidityMap,
+    IConditionsCalloutProps,
+    IConditionsCalloutStyleProps,
+    IConditionsCalloutStyles
+} from './ConditionsCallout.types';
+import { getStyles } from './ConditionsCallout.styles';
+import {
+    classNamesFunction,
+    useTheme,
+    styled,
+    Callout,
+    Stack,
+    DirectionalHint,
+    TextField,
+    PrimaryButton,
+    DefaultButton
+} from '@fluentui/react';
+import ConditionSummary from './ConditionSummary';
+import ActionItem from './ActionItem';
+import { ConditionCalloutReducer } from './ConditionsCallout.state';
+import { IDTDLPropertyType } from '../../../../../../Models/Types/Generated/3DScenesConfiguration-v1.0.0';
+import { useTranslation } from 'react-i18next';
+import { areRangesValid, checkValidity } from './ConditionCalloutUtility';
+import { deepCopy } from '../../../../../../Models/Services/Utils';
+import { CalloutInfoType } from '../ConditionsList.types';
+import { ActionItemKey } from './ActionItem.types';
+
+const getClassNames = classNamesFunction<
+    IConditionsCalloutStyleProps,
+    IConditionsCalloutStyles
+>();
+
+const ROOT_LOC = '3dSceneBuilder.visualRuleForm';
+const LOC_KEYS = {
+    addCondition: `${ROOT_LOC}.addCondition`,
+    editCondition: `${ROOT_LOC}.editCondition`,
+    labelField: `${ROOT_LOC}.labelField`
+};
+
+const ConditionsCallout: React.FC<IConditionsCalloutProps> = (props) => {
+    const {
+        calloutType,
+        onDismiss,
+        onSave,
+        target,
```

Comment:
When I click on Add new rule, the flyout appears all the way to the right, was it intended? Cant we make it appear close to target which is Add rule action button?

Also see the values prefilled with "undefined", was it also intended?

Last thing, is the order of Save, cancel buttons. As far as I remember the pattern we use is secondary, primary
![image](https://user-images.githubusercontent.com/45217314/198154636-3bc64846-cd60-4fba-987b-cd4432778a49.png)

Reply:
ah also with the edit condition callout, the beak shows a little upward of the selected list item, can we center it with the selected list items overflow menu?:
![image](https://user-images.githubusercontent.com/45217314/198156277-31b78805-7856-4d7e-a6e2-b7bba84b318a.png)

Reply:
I will fix this in a future PR, there needs to be an update to cardboard list to allow id so I can target the item as the callout target. Good catch on the 'new condition' button, I'll fix that as well in the PR.

Reply:
First thing got fixed

Reply:
sounds good!